coderd/workspaceagents.go

```diff
@@ -285,38 +277,36 @@ func (api *API) patchWorkspaceAgentStartupLogs(rw http.ResponseWriter, r *http.R
 			return
 		}
 		level = append(level, parsedLevel)
-		eof = append(eof, logEntry.EOF)
 	}
 
 	var logs []database.WorkspaceAgentStartupLog
-	// Ensure logs are not written after EOF.
-	eofError := xerrors.New("EOF log already received")
+	// Ensure logs are not written after script ended.
+	scriptEndedError := xerrors.New("startup script has ended")
 	err := api.Database.InTx(func(db database.Store) error {
-		isEOF, err := db.GetWorkspaceAgentStartupLogsEOF(ctx, workspaceAgent.ID)
+		ss, err := db.GetWorkspaceAgentLifecycleStateByID(ctx, workspaceAgent.ID)
```

Comment:
nit: scriptState :)?

Reply:
I prefer short variables names for short blocks of code, as per https://github.com/golang/go/wiki/CodeReviewComments#variable-names, but I can change this if you prefer. However, `ss` is wrong here and left-over from an earlier approach to the fix, I would call it `ls` now. 😆

Reply:
I can't argue about this as we don't have conventions, more like personal preferences, but `state` > `ls` :)

Reply:
Sure, `state` works 👍🏻.